### PR TITLE
add channel categories

### PR DESCRIPTION
**Pull request by devsnek**
**Please describe the changes this PR makes and why it should be merged:**
wowee channel categories very exiciting

**Semantic versioning classification:**  
- [x] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/structures/GuildChannel.js [hunk 1]
```diff
@@ -37,6 +37,12 @@ class GuildChannel extends Channel {
     this.position = data.position;
 
     /**
+     * The ID of the category parent of this channel
+     * @type {?snowflake}
```
Comment (PgBiel) [inline src/structures/GuildChannel.js]: Snowflake*

File: src/structures/Guild.js [hunk 2]
```diff
@@ -920,10 +919,9 @@ class Guild {
   /**
    * Creates a new channel in the guild.
    * @param {string} name The name of the new channel
-   * @param {string} type The type of the new channel, either `text` or `voice`
-   * @param {Object} [options={}] Options
```
Comment (PgBiel) [inline src/structures/Guild.js]: Shouldn't this be optional once again?

Comment (appellation) [inline src/structures/Guild.js]: ...it is optional?

Comment (PgBiel) [inline src/structures/Guild.js]: he made it not optional in the docs, see the lines after it @appellation

Comment (PgBiel) [inline src/structures/Guild.js]: also he just fixed it

**Comment by codingJWilliams** [comment]
Is there any documentation for this?

**Pull request merged**